include/reactor-uc/schedulers/dynamic/scheduler.h

```diff
@@ -0,0 +1,30 @@
+//
+// Created by tanneberger on 11/16/24.
+//
+
+#ifndef SCHEDULER_H
+#define SCHEDULER_H
+#include "reactor-uc/error.h"
+#include "reactor-uc/queues.h"
+#include "reactor-uc/scheduler.h"
+
+typedef struct DynamicScheduler DynamicScheduler;
+typedef struct Environment Environment;
+
+struct DynamicScheduler {
+  Scheduler scheduler;
+  Environment *env;
```

Comment:
This could go on the super struct since it is basically needed by all schedulers